### DISCUSSION TITLE
[@types/semver] `semver` does not respect constraints containing a comma for version ranges

### DISCUSSION THREAD

**Discussion by boesing** [DefinitelyTyped/DefinitelyTyped#61669] [Issues with a @types package] [answered]
Hey there,

we are using the `semver` typescript implementation to actually verify some very specific constraints.
We stumbled over this constraint: `>=5.6,<=8.1.99`

This actually allows all versions starting from `5.6.0` up to `8.1.99`.

Sadly, that constraint cannot be verified when using this package.

References to other semver implementations which do support `,` and thus is assumed to be a valid constraint:
- https://pkg.go.dev/github.com/masterminds/semver#hdr-Basic_Comparisons
- https://getcomposer.org/doc/articles/versions.md#version-range
- https://docs.renovatebot.com/modules/versioning/#cargo-versioning

Is this something this package is just missing or did you decide to ignore these quirks (not sure if that style is actually defined within the semver 2.0 spec).

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment, marked as answer]
> Sadly, that constraint cannot be verified when using this package.

you meant native 'semver' obviously. The DT package is just TS syntax to enable strict typings for `semver`, no actual implementation changes

**Reply by boesing** [reply]
Dang you guys are right. Sorry! Thanks for clarification here.

**Comment by BendingBender** [comment]
Hi @boesing, it looks like you have a problem with the base package [semver](https://github.com/npm/node-semver). You should file this in their bug tracker. DefinitelyTyped only hosts the TypeScript definitions, not the package itself.
